### ISSUE TITLE
Islands cannot return null

### ISSUE THREAD

**Issue by Tims777** [denoland/fresh#1129]
Since the plugin API is rather rudimentary right now, it might be desireable to create some Islands only for their side effects, e.g.:

```
import { useEffect, } from "preact/hooks";

export default function Nothing() {
  useEffect(() => alert("I am only here for the effect"));
  return null;
}
```
However, the given example does currently result in the following error:
```
Uncaught TypeError: replaceNode[(replaceNode.length - 1)] is undefined
    createRootFragment main.ts:9
    walk main.ts:46
    walk main.ts:61
    walk main.ts:61
    walk main.ts:62
    walk main.ts:62
    revive main.ts:64
    <anonymous> (index):43
```

Returning an actual node (e.g. empty `<div/>`)  instead of `null` prevents that error but is not an ideal solution in my opinion, as unneccessary elements will end up in the DOM.

**Comment by lucacasonato** [comment]
Related #502

**Comment by iuioiua** [comment]
This issue's been fixed in #1172.
